### PR TITLE
Bid-7-user-model-implementation-for-registration

### PR DESCRIPTION
**Pull request by afm-wtag**
## Description
[BE - Implement User Registration and Login Endpoints](https://welltravel.atlassian.net/browse/BID-7)
This PR adds a User model with required fields (firstName, lastName, email, contactNumber, password) and optional fields (address, balance). The registration endpoint validates input and ensures unique email addresses. Passwords are hashed using bcrypt for security. It also introduces login functionality to authenticate users and return a secure response.

## Changes
1. User Schema
3. Endpoints
4. Password Management

### CODE REVIEW

**Comment by hhe-wtag** [comment]
I am refactoring the code little bit for the auth task, I'll rename the create to register for registering the user to make it consistent.

**Review by hhe-wtag — APPROVED** [review]
I am refactoring the code little bit for the auth task in upcoming PR, I'll rename the create to register for registering the user to make it consistent. Other than that, it looks good to me. Thanks!

File: backend/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   },
   "homepage": "https://github.com/hhe-wtag/bidify#readme",
   "dependencies": {
+    "bcryptjs": "^2.4.3",
```
Comment (hhe-wtag) [inline backend/package.json:25]: We should use updated `bcrypt` version. Use this package instead please https://www.npmjs.com/package/bcrypt

**Review by hhe-wtag — DISMISSED** [review]
I requested change on the bcrypt package. To use `bcrypt` instead `bcryptjs`

Edit: Dismissed request with a comment below. 

Another Edit: Again requesting to change. (I looked deeper into the discussions for `bcryptjs`). Sorry for inconvenience.

File: backend/package.json (hunk 1 above)
Comment (hhe-wtag) [inline backend/package.json]: This version of bcrypt is built with JavaScript and last updated version was made 8 years ago. I am suggesting to use https://www.npmjs.com/package/bcrypt which is built with C++ resulting in faster and more efficient.

**Review by hhe-wtag — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 1 above)
Comment (aht-wtag) [reply backend/package.json]: @hhe-wtag why are we using bcryptjs then instead of bcrypt? Was it specifically mentioned to use this in the requirements?

Comment (hhe-wtag) [reply backend/package.json]: @aht-wtag no bhaia there was no instruction on that. This PR is made by Mumu. I requested to change `bcryptjs` to `bcrypt` instead.

Comment (afm-wtag) [reply backend/package.json]: I just overlooked the package as it was functional i worked with it. I am updating my code as bycrpt is recent. Thank you!

**Review by hhe-wtag — APPROVED** [review]

**Comment by aht-wtag** [comment]
@afm-wtag apu, just for clarification, through out this PR, you stopped using asyncHandler which @hhe-wtag apu introduced. is there any specific reason for that?

File: backend/src/app.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ app.use(
 app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
-app.use('/api/v1/users', UserRoutes);
+app.use('/api/auth', UserRoutes);
 
-app.use(globalErrorHandler);
+// app.use(globalErrorHandler);
```
Comment (aht-wtag) [inline backend/src/app.js]: Do you plan to use this later on? if not, can we get rid of this un-necessary comment?

Comment (afm-wtag) [reply backend/src/app.js]: I commented it out earlier due to an unintuitive error in the terminal that caused difficulties. Later, Hena Apu updated the global error handler, but I forgot to uncomment it. I've now resolved the issue. Thank you.

**Pull request merged**